File: backend/src/main/java/com/penocean/ehs/model/WorkerVoice.java
package com.penocean.ehs.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkerVoice {

    private Long id;
    private String voiceNo;                 // WV-YYYYMMDD-NNNN
    private String voiceType;               // NEAR_MISS / INCIDENT / INQUIRY
    private String title;
    private String content;
    private Long companyId;
    private Long vesselId;
    private Long reporterUserId;
    private Boolean reporterAnonymous;
    private String severity;                // LOW / MEDIUM / HIGH / CRITICAL
    private String status;                  // SUBMITTED / TRIAGED / IN_PROGRESS / RESOLVED / CLOSED
    private Long assignedTo;
    private LocalDateTime resolvedAt;
    private String resolution;
    private String emailSentTo;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean deleted;
}

File: frontend/src/types/workerVoice.ts
// Backend: /api/worker-voices
import type { PageResponse } from './common'

export type WorkerVoiceType = 'NEAR_MISS' | 'INCIDENT' | 'INQUIRY'
export type WorkerVoiceSeverity = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'
export type WorkerVoiceStatus =
  | 'SUBMITTED'
  | 'TRIAGED'
  | 'IN_PROGRESS'
  | 'RESOLVED'
  | 'CLOSED'

export interface WorkerVoiceAttachment {
  id: number
  fileName: string
  filePath: string
  fileSize: number
  mimeType: string
  uploadedAt: string
}

export interface WorkerVoiceListItem {
  id: number
  voiceNo: string
  voiceType: WorkerVoiceType
  title: string
  companyId: number
  companyName: string
  vesselId?: number | null
  vesselName?: string | null
  severity?: WorkerVoiceSeverity | null
  status: WorkerVoiceStatus
  reporterAnonymous: boolean
  createdAt: string
  resolvedAt?: string | null
  commentCount: number
}

export interface WorkerVoiceDetail extends WorkerVoiceListItem {
  content: string
  reporterUserId?: number | null
  reporterName?: string | null
  assignedTo?: number | null
  assigneeName?: string | null
  resolution?: string | null
  resolvedAt?: string | null
  updatedAt?: string | null
  emailSentTo?: string | null
  attachments: WorkerVoiceAttachment[]
}

export interface WorkerVoiceCreatePayload {
  voiceType: WorkerVoiceType
  title: string
  content: string
  companyId?: number | null
  vesselId?: number | null
  severity?: WorkerVoiceSeverity | null
  reporterAnonymous: boolean
}

export interface WorkerVoiceStatusPayload {
  status: WorkerVoiceStatus
  assignedTo?: number | null
  resolution?: string | null
}

export interface WorkerVoiceListParams {
  voiceType?: WorkerVoiceType | ''
  status?: WorkerVoiceStatus | ''
  keyword?: string
  page?: number
  size?: number
}

export type WorkerVoicePage = PageResponse<WorkerVoiceListItem>
